docs/content/ingestion/index.md

```diff
@@ -100,8 +100,9 @@ This is for realtime ingestion. Make sure to include "io.druid.extensions:druid-
 | type | String | This should say `avro_stream`. | no |
 | avroBytesDecoder | JSON Object | Specifies how to decode bytes to Avro record. | yes |
 | parseSpec | JSON Object | Specifies the format of the data. | yes |
+| schemaMappings | JSON Object | Specifies a mapping from field names found within the parseSpec to data within a complex avro structure. | no |
```

Comment:
@gianm, @himanshug, @zhaown 
I agree it would be nice if similar mechanisms could be used for these two very similar operations, and I think we should try to bring them together over time if possible. But currently I see a couple of issues that are impeding this.

1. A similar, suitable Xpath-like path parsing library isn't availble yet for avro based on a quick internet search. There has been some discussion in the avro community about building such a thing, and a scala-based implementation does exist, but I don't see a Java version available.
2. The relationship between InputRowParsers, ParseSpecs, and Parsers makes it hard to extend the parsing in a way that can share this type of facility. Because Parsers assume the data they are parsing is text, implementing new formats is more easily done at the InputRowParser level without adding new Parsers and ParseSpecs, which means encoding this type of flattening information in the ParseSpec (as has been done with JSON) makes it hard to share with other implementations. 

That said, the syntax currently being used for these extractions for avro in this PR is fairly similar to the syntax used by JsonPath. We could tweak a couple of things, like forcing all expressions to start with the `$` character to make them look a bit more like JsonPath, but I think that might just make things more confusing rather than less given that support for the whole JsonPath syntax is missing. The concepts are similar enough that I think users will understand the two mechanisms, even though there are differences. I think its a much bigger effort to move this information to a common location shared between avro and json, as that involves refactoring the whole InputRowParser/Parser/ParseSpec mess and introducing all sorts of backward incompatible changes.